src/Compilers/VisualBasic/Portable/Compilation/MethodCompiler.vb

```diff
@@ -1193,7 +1194,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                 End If
                                 Return semanticModel
                             End Function)
-                        compilation.EventQueue.Enqueue(New SymbolDeclaredCompilationEvent(compilation, method, lazySemanticModel))
+                        If method.Name = "PartialMethod" Then
+                            method = method
+                        End If
+                        Dim symbolToProduce = If(method.PartialDefinitionPart, method)
+                        compilation.EventQueue.Enqueue(New SymbolDeclaredCompilationEvent(compilation, symbolToProduce, lazySemanticModel))
```

Comment:
The same issue with possible inconsistency between the symbol and semantic model.